### PR TITLE
20738: implemented controller cert domain config

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	certDomain := d.Get("cert_domain").(string)
+
+	err := client.SetCertDomain(ctx, certDomain)
+	if err != nil {
+		return diag.Errorf("could not set cert domain: %v", err)
+	}
+
+	d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+	return nil
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: Should return resourceAviatrixControllerCertDomainConfigRead

File: aviatrix/resource_aviatrix_controller_cert_domain_config_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccControllerCertDomainConfigBasic(rName string) string {
+	return fmt.Sprintf(`
+resource "aviatrix_controller_cert_domain_config" "test" {
+  cert_domain = "%s"
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_controller_cert_domain_config_test.go]: Should be a four-character indent.

File: docs/resources/aviatrix_controller_cert_domain_config.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+# aviatrix_controller_cert_domain_config
+
+The **aviatrix_controller_cert_domain_config** resource allows management of an Aviatrix Controller's cert domain config. This resource is available as of provider version R2.19+.
+
+## Example Usage
+
+```hcl
+# Create an an Aviatrix controller cert domain config
```
Comment (zhiqiangjin) [inline docs/resources/aviatrix_controller_cert_domain_config.md]: Remove the extra "an"

File: docs/resources/aviatrix_controller_cert_domain_config.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  cert_domain = "abc.com"
+}
+```
+
+
+## Argument Reference
+
+The following argument is supported:
+
+* `cert_domain` - (Optional) Domain name that is used in FQDN for generating cert. Default value: "aviatrixnetwork.com".
```
Comment (zhiqiangjin) [inline docs/resources/aviatrix_controller_cert_domain_config.md:27]: Is it required?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go (hunk 1 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_controller_cert_domain_config.go:43]: There's only one API call in Create(), do we still need to return Read()?

File: docs/resources/aviatrix_controller_cert_domain_config.md (hunk 4 above)
Comment (shuoliugit) [reply docs/resources/aviatrix_controller_cert_domain_config.md:27]: It's optional. I corrected it in resource_aviatrix_controller_cert_domain_config.go

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**